File: Part 2 - Regression/Polynomial Regression/Polynomial_Reg.py
#importing libraries
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt

#importing dataset
data=pd.read_csv("Position_Salaries.csv")
X=data.iloc[:,1:2].values
y=data.iloc[:,2].values


#LinerReg model building
from sklearn.linear_model import LinearRegression
lin_reg=LinearRegression()
lin_reg.fit(X,y)

#BUilding a polynomial Reg model
from sklearn.preprocessing import PolynomialFeatures
ploy_fea=PolynomialFeatures(degree=4)
X_ply=ploy_fea.fit_transform(X)
poly_reg=LinearRegression()
poly_reg.fit(X_ply,y)

#visualising the data in both liner and polynomial
X_grid=np.arange(min(X),max(X),0.1)
X_grid=X_grid.reshape(len(X_grid),1)

plt.scatter(X,y,color="red")
plt.plot(X,lin_reg.predict(X),color="blue")
plt.plot(X_grid,poly_reg.predict(ploy_fea.fit_transform(X_grid)),color="black")
plt.title("Salary Detect")
plt.xlabel("Years Of Exp")
plt.ylabel("Salary")

File: Part 2 - Regression/Polynomial Regression/Polynomial_Reg.R
#importing dataset
data=read.csv("Position_Salaries.csv")
data=data[2:3]

#building linear model
lin_reg=lm(formula=Salary ~ .,data = data)

#building a polynomial reg
data$Level2=data$Level^2
data$Level3=data$Level^3
data$Level4=data$Level^4

ploy_reg=lm(formula = Salary ~ .,data=data)

#visualising the data
library(ggplot2)

ggplot()+
  geom_point(aes(x=data$Level,y=data$Salary),color="blue")+
  geom_line(aes(x=data$Level,y=predict(lin_reg,newdata = data)),color="red")+
  geom_line(aes(x=data$Level,y=predict(ploy_reg,newdata = data)),color="green")+
  ggtitle("Salary Detect")+
  xlab("Expericence")+
  ylab("Salary")